engine/channels.go

```diff
@@ -24,6 +24,20 @@ func RolesByChannelID(channelID string) (flow.RoleList, bool) {
 	return roles, ok
 }
 
+// ValidChannelID returns true if channel ID is a valid channel, i.e.,
+// it is either a cluster channel id or is registered in the channel id map.
+func ValidChannelID(channelID string) bool {
```

Comment:
What would happen if someone added a new channel ID but forgot to add it to `channelIdMap`?

Reply:
The register method would throw an error and would not allow the engine to register it. It is very similar to the case where someone generates a new message type and forgot to add its encoder and decoder.

Reply:
> I wonder if we can make it more difficult to make that mistake, or make it easy to detect. 
Currently it looks like this:
```
	if !channels.ValidChannelID(channelID) {
		return nil, fmt.Errorf("unknown channel id: %s, should be registered in topic map", channelID)
	}
```

However, we can of course make it automated (i.e., registering a channel automatically adds it to the id map). The challenge is the role heterogeneity, which impedes all nodes to have the same id map. So, currently, we stick to the low-hanging fruit and add it statically. 
cc @vishalchangrani